lib/src/mobile_scanner.dart

```diff
@@ -270,33 +280,37 @@ class _MobileScannerState extends State<MobileScanner>
       );
     }
     if (controller.autoStart) {
-      controller.start();
+      await controller.start();
     }
-    super.initState();
   }
 
-  @override
-  void dispose() {
-    super.dispose();
-
-    if (_subscription != null) {
-      _subscription!.cancel();
-      _subscription = null;
-    }
+  Future<void> disposeMobileScanner() async {
+    await _subscription?.cancel();
```

Comment:
I would move the WidgetsBinding observer call to the font? Also, null-out the subscription?

```suggestion
  Future<void> disposeMobileScanner() async {
    if (widget.controller == null) {
      WidgetsBinding.instance.removeObserver(this);
    }
  
    await _subscription?.cancel();
    _subscription = null;
    
    if (controller.autoStart) {
      await controller.stop();
    }
    
    // Dispose default controller if not provided by user
    if (widget.controller == null) {
      await controller.dispose();
    }    
  }
```

Reply:
I didn't null it out, because in the `didChangeAppLifecycleState` function, `case AppLifecycleState.inactive:` also doesn't null it out because of syncronous commands. However, maybe we need to create a separate async function for that aswell.

Ultimately it's not really a good pattern of using async methods in `dispose` and `didChangeAppLifecycleState`, but because we are handling a lot on the native side, there isn't really any better option for now. We can look into it for the new major release, or even the next one after that.

Reply:
Yeah, async dispose/initState/didChangeAppLifecycleState is meh. So I'd rather go with wrapping them in `unawaited` then.